openhcl/underhill_core/src/dma_manager.rs

```diff
@@ -0,0 +1,85 @@
+use std::sync::{Arc, Mutex, Weak};
+use memory_range::MemoryRange;
+use once_cell::sync::OnceCell;
+
+pub use dma_client::{DmaClient, DmaInterface, DmaTransaction, DmaTransactionHandler};
+
+pub enum DmaError {
+    InitializationFailed,
+    MapFailed,
+    UnmapFailed,
+    PinFailed,
+    BounceBufferFailed,
```

Comment:
Please use source attributes so that we don't lose error origination. E.g.

```
#[derive(Error, Debug)]
pub enum DmaError {
...
PinFailed(#[source] ... error type)
```